roles/tls_install_certs/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Check if signed cert is available
+  become: no
+  delegate_to: localhost
+  stat:
+    path: "{{ local_tls_signed_certs_dir }}/{{ inventory_hostname }}.pem"
+  register: signed_cert
+
+- ansible.builtin.fail:
+    msg: >
+      "Signed cert for {{ inventory_hostname }} could not be found. If manual signing is
+      required, do this now and re-run the playbook with 'tls_signed_certs_dir' variable set.
+  when:
+    - not signed_cert.stat.exists
+
+- name: Copy signed certs to hosts
+  ansible.builtin.copy:
+    src: "{{ local_tls_signed_certs_dir }}/{{ inventory_hostname }}.pem"
+    dest: "{{ tls_cert_path }}"
+    mode: 0644
+
+- name: Create host agnostic link for signed certificate
+  ansible.builtin.file:
+    src: "{{ tls_cert_path }}"
+    dest: "{{ tls_cert_path_generic }}"
+    state: hard
+    force: true
+    mode: 0644
+    owner: root
+    group: root
+
+- name: Validate certificate
```

Comment:
Potentially use the [community.crypto.x509_certificate_info](https://docs.ansible.com/ansible/latest/collections/community/crypto/x509_certificate_info_module.html) module?

Reply:
I tested this but it seems to not validate against the CA cert. Instead I've continued to use the `openssl verify` command but changed to module to `ansible.builtin.command`